### PR TITLE
go/signedexchange: More tests for signature verification

### PR DESCRIPTION
**Pull request by irori**

### CODE REVIEW

File: go/signedexchange/signedexchange_test.go [hunk 1]
```diff
@@ -227,13 +227,15 @@ func TestSignedExchangeBannedCertUrlScheme(t *testing.T) {
 	}
 }
 
-func TestVerify(t *testing.T) {
+func createTestExchange(t *testing.T) (e *Exchange, s *Signer, certBytes []byte) {
 	u, _ := url.Parse("https://example.com/")
-	header := http.Header{}
-	header.Add("Content-Type", "text/html; charset=utf-8")
+	reqHeaders := http.Header{}
+	reqHeaders.Add(":method", "GET")
```
Comment (nyaxt) [inline go/signedexchange/signedexchange_test.go]: The current situation around `Exchange.RequestHeaders` is quite tricky. We might want to resolve it first.
`func (*Exchange) encodeRequest()` assumes `":method"` key to not be in the `RequestHeaders` map, while `func (*Exchange) decodeRequest()` will populate the key.

Comment (irori) [reply go/signedexchange/signedexchange_test.go]: OK, created #344.

**Comment by irori** [comment]
Rebased onto master after #344. PTAL

File: go/signedexchange/signedexchange_test.go [hunk 2]
```diff
@@ -229,7 +229,7 @@ func TestSignedExchangeBannedCertUrlScheme(t *testing.T) {
 	}
 }
 
-func TestVerify(t *testing.T) {
+func createTestExchange(t *testing.T) (e *Exchange, s *Signer, certBytes []byte) {
```
Comment (nyaxt) [inline go/signedexchange/signedexchange_test.go]: @hajimehoshi Any thoughts on using named return values here?
https://tour.golang.org/basics/7 discourages using it on long func, but I'm open here.

**Pull request merged**